examples/vision/attention_mll_classification.py

```diff
@@ -0,0 +1,579 @@
+    neg_count = len(np.where(labels == 0)[0])
+    pos_count = len(np.where(labels == 1)[0])
+    total_count = neg_count + pos_count
+
+    # Build class weight dictionary.
+    return {
+        0: (1 / neg_count) * (total_count / 2),
+        1: (1 / pos_count) * (total_count / 2),
+    }
+
+
+"""
+## Build and Train Model
+
+The model is built and trained in this section.
+
+Some regularization techniques are considered to avoid overfitting the model which ensures
+minimal generalization error.
+"""
+
+
+def train(train_data, train_labels, val_data, val_labels, model):
+
+    # Train model.
+    # Prepare callbacks.
+    # Path where to save best weights.
+    with tempfile.NamedTemporaryFile(suffix=".h5") as tmp:
+
+        # Take the file name from the wrapper.
+        file_path = tmp.name
+
+        # Initialze model checkpoint callback.
+        mc = callbacks.ModelCheckpoint(
+            file_path,
+            monitor="val_loss",
+            verbose=0,
+            mode="min",
+            save_best_only=True,
+            save_weights_only=True,
+        )
+
+        # Initialze early stopping callback.
+        # The model performance is monitored across the unseen data and stops training
+        # when the generalization error cease to decrease
+        es = callbacks.EarlyStopping(monitor="val_loss", patience=10, mode="min")
+
+        # Compile model.
+        model.compile(
+            optimizer="adam",
+            loss="sparse_categorical_crossentropy",
+            metrics=["accuracy"],
+        )
+
+        # Fit model.
+        model.fit(
+            train_data,
+            train_labels,
+            validation_data=(val_data, val_labels),
+            epochs=20,
+            class_weight=calculate_weights(train_labels),
+            batch_size=1,
+            callbacks=[es, mc],
+            verbose=0,
+        )
+
+        # Load best weights.
+        model.load_weights(file_path)
+
+        return model
+
+
+# Building model(s).
+instance_shape = train_data[0][0].shape
+models = [create_model(instance_shape) for _ in range(ENSEMBLE_AVG_COUNT)]
+
+# Training model(s).
+trained_models = [
+    train(train_data, train_labels, val_data, val_labels, model)
+    for model in tqdm(models)
+]
+
+"""
+# Model Evaluation
+
+The models are in their deterministic state now and ready for evaluation.
+
+Based on the number of models (`ENSEMBLE_AVG_COUNT`), the models predict the results
+and then averaged out (equal contribution per model).
+"""
+
+
+def predict(data, labels, trained_models):
+    predictions = []
+    attention_weights = []
+    losses = []
+    accuracies = []
+
+    for model in trained_models:
+
+        # Predict output classes on validation data.
+        pred = model.predict(data)
+        predictions.append(pred)
+
+        # Create intermediate model to get attention layer weights.
+        interm_model = keras.Model(model.input, model.get_layer("alpha").output)
```

Comment:
This leads to creating lots of models each time you call the function, but in reality you only need to create each model once.

Reply:
I am not sure I understood this comment.
I will explain what I am doing here and please let me know if you have other suggestions.

Based on the `ENSEMBLE_AVG_COUNT`, I create number of models. For each model, at inference or prediction mode, I create an intermediate model (as of here https://keras.io/getting_started/faq/#how-can-i-obtain-the-output-of-an-intermediate-layer-feature-extraction) to get the weights of the attention layer. Where as  the complete model will output the class probabilities.
Hence, if my `ENSEMBLE_AVG_COUNT` was 3, I will create other 3 intermediate models.

Shall I keep it this way or there is a better alternative?